components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dao/impl/IdentityUserMetadataMgtDAOImpl.java

```diff
@@ -0,0 +1,508 @@
+                    StringBuilder data = new StringBuilder("{");
+                    if (userIdentityDTO.getUserIdentityDataMap() != null) {
+                        for (Map.Entry<String, String> entry : userIdentityDTO.getUserIdentityDataMap().entrySet()) {
+                            data.append("[").append(entry.getKey()).append(" = ")
+                                    .append(entry.getValue()).append("], ");
+                        }
+                    }
+                    if (data.indexOf(",") >= 0) {
+                        data.deleteCharAt(data.lastIndexOf(","));
+                    }
+                    data.append("}");
+                    log.debug("Storing UserIdentityClaimsDO to cache for user: " + userName + " with claims: " + data);
+                }
+
+                String domainName = userStoreManager.getRealmConfiguration().
+                        getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+
+                IdentityDataStoreCacheKey key = new IdentityDataStoreCacheKey(domainName, userName);
+                int tenantId = userStoreManager.getTenantId();
+                UserIdentityClaim cachedUserIdentityDTO = identityDataStoreCache.getValueFromCache(key, tenantId);
+                if (cachedUserIdentityDTO != null) {
+                    cachedUserIdentityDTO.getUserIdentityDataMap().putAll(userIdentityDTO.getUserIdentityDataMap());
+                    identityDataStoreCache.addToCache(key, cachedUserIdentityDTO, tenantId);
+                } else {
+                    identityDataStoreCache.addToCache(key, userIdentityDTO, tenantId);
+                }
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            log.error("Error while obtaining tenant ID from user store manager", e);
+        }
+    }
+
+    /**
+     * Store user metadata in the user store.
+     *
+     * @param userStoreManager          User store manager
+     * @param username                  Username
+     * @param tenantId                  Tenant ID
+     * @param claimURI                  Claim URI
+     * @param value                     Claim attribute value
+     * @param eventName                 Event name
+     * @throws IdentityEventException   Identity event exception
+     */
+    private void storeMetadataToUserStore(UserStoreManager userStoreManager, String username, String tenantId,
+                                          String claimURI, String value, String eventName)
+            throws IdentityEventException {
+
+        String claimAttributeName = retrieveAttributeName(userStoreManager, tenantId, claimURI);
+        boolean executionResult;
+        try (Connection connection = getDBConnection(userStoreManager.getRealmConfiguration())) {
+            try {
+                if (isMetadataExistsInUserStore(connection, userStoreManager, tenantId, username, claimAttributeName)) {
+                    // update claim in user store
```

Comment:
The comment is not formatted properly. Start with an uppercase letter and end with a full-stop. Check other places as well.

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/697/commits/dd3419fd69cc40e847e40ed9dfb29c867a3f0018